### PR TITLE
CIRC-1711: Cancel actual cost lost item fees when declaring item lost

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
## Purpose
Existing open actual cost record (lost item fee with actual cost) for the loan must be cleared when declaring item lost.

Resolves [CIRC-1711](https://issues.folio.org/browse/CIRC-1711)

#### TODOS and Open Questions
- [x] Check logging

### CODE REVIEW

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java [hunk 1]
```diff
@@ -114,44 +114,47 @@ public CompletableFuture<Result<RenewalContext>> refundLostItemFees(
...
       .thenCompose(r -> r.after(this::cancelActualCostFee))
       .thenApply(r -> r.map(ignored -> context));
   }
 
   public CompletableFuture<Result<LostItemFeeRefundContext>> refundAccounts(
     LostItemFeeRefundContext context) {
 
+    Collection<Account> accounts = context.getLoan().getAccounts();
+
+    if (accounts == null || accounts.isEmpty()) {
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java:150]: CollectionUtils.isEmpty ?

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java [hunk 2]
```diff
@@ -114,44 +114,47 @@ public CompletableFuture<Result<RenewalContext>> refundLostItemFees(
...
   }
 
-  public CompletableFuture<Result<LostItemFeeRefundContext>> processRefund(
+  private CompletableFuture<Result<LostItemFeeRefundContext>> processRefund(
     LostItemFeeRefundContext context) {
 
-    log.debug("processRefund:: loanId={}, itemId={}, cancelReason={}",
-      context::getLoanId, context::getItemId, context::getCancelReason);
+    if (!context.getLostItemPolicy().shouldRefundFees(context.getItemLostDate())) {
+      log.info("refundLostItemFees:: refund interval was exceeded for loan {}", context::getLoanId);
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java:135]: method name processRefund

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java [hunk 3]
```diff
@@ -114,44 +114,47 @@ public CompletableFuture<Result<RenewalContext>> refundLostItemFees(
...
     LostItemFeeRefundContext context) {
 
+    Collection<Account> accounts = context.getLoan().getAccounts();
+
+    if (accounts == null || accounts.isEmpty()) {
+      log.info("No accounts to refund for loan {}", context.getLoanId());
+      return ofAsync(context);
+    }
+
+    log.info("Refunding {} accounts for loan {}: {}", accounts::size, context::getLoanId,
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java:155]: should there be method name?

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java [hunk 4]
```diff
@@ -114,44 +114,47 @@ public CompletableFuture<Result<RenewalContext>> refundLostItemFees(
   public CompletableFuture<Result<LostItemFeeRefundContext>> refundLostItemFees(
     LostItemFeeRefundContext refundFeeContext) {
 
-    log.debug("refundLostItemFees:: loanId={}, itemId={}, cancelReason={}",
-      refundFeeContext::getLoanId, refundFeeContext::getItemId, refundFeeContext::getCancelReason);
+    log.info("refundLostItemFees::Attempting to refund lost item fees: loanId={}, cancelReason={}",
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java:117]: missed space after ::

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java (hunk 1 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java]: Sure, why not.

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java (hunk 2 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java]: fixed, thanks

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java (hunk 3 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java]: yes, fixed

File: src/main/java/org/folio/circulation/services/LostItemFeeChargingService.java [hunk 5]
```diff
@@ -72,52 +71,53 @@ public LostItemFeeChargingService(Clients clients,
...
+      return ofAsync(referenceDataContext.withLostItemPolicy(lostItemPolicy));
     }
 
-    if (!hasLostItemFees(loan) && shouldCloseLoan(referenceData.lostItemPolicy)) {
-      log.info("No existing lost item fees found, closing loan [{}] as lost and paid.", loan.getId());
+    return lostItemPolicyRepository.getLostItemPolicyById(loan.getLostItemPolicyId())
+      .thenApply(r -> r.map(referenceDataContext::withLostItemPolicy));
+  }
+
+  private CompletableFuture<Result<Loan>> applyLostItemFeePolicy(ReferenceDataContext context) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/services/LostItemFeeChargingService.java:111]: Log method start

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java [hunk 6]
```diff
@@ -124,34 +125,37 @@ public CompletableFuture<Result<LostItemFeeRefundContext>> refundLostItemFees(
...
-        return fetchAccountsAndActionsForLoan(context)
-          .thenCompose(r -> r.after(this::processRefund))
-          .exceptionally(CommonFailures::failedDueToServerError);
-      }));
+      .thenCompose(r -> r.after(this::processRefund))
+      .exceptionally(CommonFailures::failedDueToServerError);
   }
 
-  public CompletableFuture<Result<LostItemFeeRefundContext>> processRefund(
+  private CompletableFuture<Result<LostItemFeeRefundContext>> processRefund(
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java:132]: Log method start

File: src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java [hunk 7]
```diff
@@ -124,34 +125,37 @@ public CompletableFuture<Result<LostItemFeeRefundContext>> refundLostItemFees(
...
+    }
 
-    return refundAccounts(context)
+    return fetchAccountsAndActionsForLoan(context)
+      .thenCompose(r -> r.after(this::refundAccounts))
       .thenCompose(r -> r.after(this::cancelActualCostFee))
       .thenApply(r -> r.map(ignored -> context));
   }
 
   public CompletableFuture<Result<LostItemFeeRefundContext>> refundAccounts(
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/services/LostItemFeeRefundService.java:146]: Log method start

**Pull request merged**